### PR TITLE
refactor: use v2 device-sdk

### LINKED ISSUE #73: Consume V2 Device SDK (REST)
1. Update go.mod file to reference new v2.0.x version of module
2. Update all import paths for the module to include /v2 in the path.
3. Update Attribution.txt to add /v2 to the module reference.

See the PR for edgex-go for example:  https://github.com/edgexfoundry/edgex-go/pull/3032/files

### PR DESCRIPTION
**Pull request by hahattan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: cmd/res/profiles/sample-numeric-device.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      valueType: "Int64"
+      readWrite: "W"
+      mediaType: "text/plain"
+  - name: float
+    isHidden: true
+    description: "Float64 value"
+    properties:
+      valueType: "Float64"
+      readWrite: "W"
+      mediaType: "text/plain"
```
Comment (cloudxxx8) [inline cmd/res/profiles/sample-numeric-device.yaml:24]: add empty line

File: cmd/res/profiles/sample-json-device.yaml [hunk 2]
```diff
@@ -8,14 +8,9 @@ description: "REST Device that sends in Json"
...
-        { type: "String", readWrite: "R" }
-
-deviceCommands:
-  - name: PostJson
-    get:
-      - { operation: "get", deviceResource: "json"}
\ No newline at end of file
+      valueType: "String"
+      readWrite: "W"
+      mediaType: "application/json"
```
Comment (cloudxxx8) [inline cmd/res/profiles/sample-json-device.yaml:16]: add empty line

File: cmd/res/profiles/sample-image-device.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      valueType: "Binary"
+      readWrite: "W"
+      mediaType: "image/jpeg"
+  - name: png
+    isHidden: true
+    description: "PNG Image binary message"
+    properties:
+      valueType: "Binary"
+      readWrite: "W"
+      mediaType: "image/png"
```
Comment (cloudxxx8) [inline cmd/res/profiles/sample-image-device.yaml:23]: add empty line

File: cmd/res/devices/sample-devices.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  Labels = [ 'rest', 'binary', 'image' ]
+  [DeviceList.Protocols]
+    [DeviceList.Protocols.other]
+[[DeviceList]]
+  Name = 'sample-numeric'
+  ProfileName = 'sample-numeric'
+  Description = 'RESTful Device that sends in numeric data'
+  Labels = [ 'rest', 'numeric', 'float', 'int' ]
+  [DeviceList.Protocols]
+    [DeviceList.Protocols.other]
```
Comment (cloudxxx8) [inline cmd/res/devices/sample-devices.toml:22]: add empty line

File: cmd/res/configuration.toml [hunk 5]
```diff
@@ -39,31 +73,9 @@ Type = 'consul'
...
-  [DeviceList.Protocols]
-    [DeviceList.Protocols.other]
-[[DeviceList]]
-  Name = 'sample-numeric'
-  Profile = 'sample-numeric'
-  Description = 'RESTful Device that sends in numeric data'
-  Labels = [ 'rest', 'numeric', 'float', 'int' ]
-  [DeviceList.Protocols]
-    [DeviceList.Protocols.other]
+[Driver]
```
Comment (cloudxxx8) [inline cmd/res/configuration.toml:81]: add empty line

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**